tests/test-stake-pool-sdk.ts

```diff
@@ -143,197 +145,270 @@ describe("test basic functionalities", () => {
+    expect(referralFeePaid.toNumber()).to.eq(inverseReferralFeePaid.toNumber());
+  });
+
+  it("it calcWithdrawals() and calcWithdrawalsInverse() works correctly", async () => {
+    const socean = new Socean("testnet");
+    const stakePool = await socean.getStakePoolAccount();
+    const validatorList = await socean.getValidatorListAccount(
+      stakePool.account.data.validatorList,
+    );
+
+    const withdrawAmountDroplets = new Numberu64(
+      Math.round(
+        Math.random() * stakePool.account.data.poolTokenSupply.toNumber(),
+      ),
+    );
+
+    const validatorWithdrawalReceipts = await calcWithdrawals(
+      withdrawAmountDroplets,
+      stakePool,
+      validatorList.account.data,
+    );
+
+    const totalLamportsReceived: Numberu64 = totalWithdrawLamports(
+      validatorWithdrawalReceipts,
+    );
+
+    const validatorWithdrawalReceiptsInverse = await calcWithdrawalsInverse(
+      totalLamportsReceived,
+      stakePool,
+      validatorList.account.data,
+    );
 
-    it("it calcSolDeposit() matches actual droplets received", async () => {
-      const socean = new Socean("testnet");
-      const stakePool = await socean.getStakePoolAccount();
-      let scnSolAtaAcctInfo = await scnSolToken.getAccountInfo(scnSolAtaPubkey);
-      const initialScnSolBalance = scnSolAtaAcctInfo.amount;
-
-      // check using a random deposit of 0-0.25 SOL
-      const depositAmountSol = Math.random() / 4;
-      const depositAmount = Math.round(depositAmountSol * LAMPORTS_PER_SOL);
-      const depositAmountLamports = new Numberu64(depositAmount);
-      const { dropletsReceived } = calcSolDeposit(
-        depositAmountLamports,
-        stakePool.account.data,
+    let totalDropletsToUnstake: Numberu64 = new Numberu64(0);
+    let totalLamportsReceivedInverse: Numberu64 = new Numberu64(0);
+
+    // Accumulating two values, so just used forEach
+    validatorWithdrawalReceiptsInverse.forEach((receipt) => {
+      totalDropletsToUnstake = Numberu64.cloneFromBN(
+        totalDropletsToUnstake.add(receipt.withdrawalReceipt.dropletsUnstaked),
       );
-      // TODO: if an epoch boundary crosses at this point
-      // and depositSol() updates the stake pool, the new supply would not match and this test will fail...
-      const lastDepositTxId = (
-        await socean.depositSol(staker, depositAmountLamports)
-      )
-        .pop()
-        ?.pop();
-      // wait until the last tx (deposit) is finalized
-      await connection.confirmTransaction(lastDepositTxId!, "finalized");
-
-      scnSolAtaAcctInfo = await scnSolToken.getAccountInfo(scnSolAtaPubkey);
-      expect(scnSolAtaAcctInfo.amount.toNumber()).to.eq(
-        initialScnSolBalance.add(dropletsReceived).toNumber(),
+      totalLamportsReceivedInverse = Numberu64.cloneFromBN(
+        totalLamportsReceivedInverse.add(
+          receipt.withdrawalReceipt.lamportsReceived,
+        ),
       );
```

Comment:
you can use `totalWithdrawLamports()` here too, no?

Reply:
I'm accumulating two values, `totalDropletsToUnstake` and `totalLamportsReceivedInverse` in a single `forEach`, and I thought it would be better to keep it together? Should I make them two diff loops, one using `totalWithdrawLamports` and one custom accumulator for `totalDropletsToUnstake` ?

Reply:
yeah i think we should create a new custom accumulator for `totalDropletsToUnstake` in our lib utils since that will probably come in handy for developing UI elements and stuff but we can do that in a different PR

Reply:
I can add it here as well, no worries